src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -513,6 +524,44 @@ private String switchToNewTab(String parentTab) {
     return newTab;
   }
 
+  private String getTabWithTitle(String title) {
+    return getTabWithCondition(nameToTitle -> nameToTitle.getValue().startsWith(title));
+  }
+
+  private String getOtherTab(String title) {
+    return getTabWithCondition(nameToTitle -> !nameToTitle.getValue().startsWith(title));
+  }
+
+  private String getTabWithCondition(
+    java.util.function.Predicate<? super Pair<String, String>> condition) {
+    Optional<String> newTab = driver
+      .getWindowHandles()
+      .stream()
+      .map(handleName -> Pair.of(handleName, driver.switchTo().window(handleName).getTitle()))
+      .filter(condition)
+      .map(Pair::getKey)
+      .findFirst();
+    return newTab.orElseThrow(() -> new NotFoundException(
+      String.format("Tab with title %1$s doesn't exist", title)));

```

Comment:
both ways are correct, yours doesn't care about the position of argument, so I think it's a better fit